web_client/dialogs/openAnnotatedImage.js

```diff
@@ -3,46 +3,104 @@ import _ from 'underscore';
 
 import { restRequest } from 'girder/rest';
 import ItemCollection from 'girder/collections/ItemCollection';
+import UserCollection from 'girder/collections/UserCollection';
 import View from 'girder/views/View';
 import 'girder/utilities/jquery/girderModal';
 
 import events from '../events';
 import router from '../router';
 
+import listTemplate from '../templates/dialogs/annotatedImageList.pug';
 import template from '../templates/dialogs/openAnnotatedImage.pug';
 import '../stylesheets/dialogs/openAnnotatedImage.styl';
 
 let dialog;
-let paths = {};
+const paths = {};
+
+const AnnotatedImageList = View.extend({
+    events: {
+        'keyup .form-control': 'fetch'
+    },
+
+    initialize() {
+        this.listenTo(this.collection, 'reset', this.render);
+    },
+
+    render() {
+        this.$el.html(listTemplate({
+            items: this.collection.toJSON(),
+            paths
+        }));
+        return this;
+    }
+});
 
 const OpenAnnotatedImage = View.extend({
     events: {
-        'click .h-annotated-image': '_submit'
+        'click .h-annotated-image': '_submit',
+        'keyup input': '_debouncedFetch',
+        'change select': '_debouncedFetch'
     },
 
     initialize() {
         this.collection = new ItemCollection();
         // disable automatic sorting of this collection
         this.collection.comparator = null;
-        this.listenTo(this.collection, 'reset', this.render);
+
+        this._users = new UserCollection();
+        this._users.sortField = 'login';
+        this._users.pageLimit = 500;
+        this._usersIsFetched = false;
+        this._users.fetch().done(() => {
+            this._usersIsFetched = true;
+            this.render();
+        });
+
+        if (createDialog.debounceTimeout > 0) {
+            this._debouncedFetch = _.debounce(this.fetch, createDialog.debounceTimeout);
```

Comment:
Debouncing by time isn't really the right way to debounce this call.  Rather, if we are in a pending fetch, we shouldn't issue another fetch until that one returns.  If the fetch is slower than the `debounceTimeout`, this could still saturate the server (and if it quicker, it won't be as responsive as it could be).